test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -13452,6 +13453,21 @@ def create_vnode_def(self, name, attrs={}, numnodes=1, sharednode=True,
         del attribs
         return "\n".join(vdef)
 
+    def add_checkpoint_script(self, dirname=None, body=None, mode=None,
+                              runas=None, sudo=False, uid=None, gid=None):
```

Comment:
I don't think you need runas, sudo, mode, uid, or gid.  The mom will complain if the script isn't secure.  I think we should force these to be root with secure permissions.